src/main/groovy/edu/oregonstate/mist/classsearchapi/dao/ClassSearchDAO.groovy

```diff
@@ -0,0 +1,184 @@
+
+        //@todo: use built-in http library: http://hc.apache.org/httpcomponents-client-ga/quickstart.html
+        def remote = new HTTPBuilder(getBackendHost())
+        remote.ignoreSSLIssues()
+        remote.auth.basic(getBackendUsername(), getBackendPassword())
+
+        remote.get( path : getBackendPath(),
+                contentType : JSON,
+                query : query) { resp, reader ->
+
+            println "response status: ${resp.statusLine}"
+            println 'Headers: -----------'
+            resp.headers.each { h ->
+                println " ${h.name} : ${h.value}"
+            }
+            data = reader
+        }
+
+        getFormattedData(data)
+    }
+
+    /**
+     * Takes the data from the backend and formats it based on the swagger spec.
+     *
+     * @param data
+     * @return
+     */
+    private static List<ResourceObject> getFormattedData(def data) {
+        List<ResourceObject> result = new ArrayList<ResourceObject>()
+
+        data.each {
+            List<Faculty> faculty = new ArrayList<Faculty>()
+            List<MeetingTime> meetingTimes = new ArrayList<MeetingTime>()
+            String status = getStatus(it)
+
+            it.faculty.each { f ->
+                faculty << new Faculty(displayName: f.displayName, primaryFaculty: f.primaryIndicator)
+            }
+            it.meetingTimes.each { k ->
+                meetingTimes << new MeetingTime(
+                        startTime:          k.beginTime,
+                        endTime:            k.endTime,
+                        building:           k.building,
+                        buildingName:       k.buildingDescription,
+                        room:               k.room,
+                        campus:             k.campus,
+                        campusDescription:  k.campusDescription,
+                        monday:             k.monday,
+                        tuesday:            k.tuesday,
+                        wednesday:          k.wednesday,
+                        thursday:           k.thursday,
+                        friday:             k.friday,
+                        saturday:           k.saturday,
+                        sunday:             k.sunday
+                )
+            }
+
+            Attributes attributes = new Attributes(
+                    campusDescription:          it.campusDescription,
+                    courseNumber:               it.courseNumber,
+                    crn:                        it.courseReferenceNumber,
+                    sectionTitle:               it.sectionTitle,
+                    creditHourHigh:             it.creditHourHigh,
+                    creditHourLow:              it.creditHourLow,
+                    creditHours:                it.creditHours,
+                    enrollment :                it.enrollment,
+                    maximumEnrollment :         it.maximumEnrollment,
```

Comment:
Is there a reason for the space before the colon? Several other attributes that are like this